pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -21,44 +22,111 @@ import (
 	"github.com/lf-edge/eve/api/go/metrics"
 	zmet "github.com/lf-edge/eve/api/go/metrics" // zinfo and zmet here
 	"github.com/lf-edge/eve/pkg/pillar/cipher"
-	"github.com/lf-edge/eve/pkg/pillar/diskmetrics"
 	"github.com/lf-edge/eve/pkg/pillar/flextimer"
 	"github.com/lf-edge/eve/pkg/pillar/types"
 	"github.com/lf-edge/eve/pkg/pillar/utils"
 	"github.com/lf-edge/eve/pkg/pillar/zedcloud"
 	uuid "github.com/satori/go.uuid"
-	"github.com/shirou/gopsutil/disk"
 	"github.com/shirou/gopsutil/host"
 )
 
-// Report disk usage for these paths
-var reportDiskPaths = []string{
-	"/",
-	types.IdentityDirname,
-	types.PersistDir,
+func handleDiskMetricStatusModify(ctxArg interface{}, key string, statusArg interface{}) {
+	status := statusArg.(types.DiskMetricStatus)
+	ctx := ctxArg.(*zedagentContext)
+	ctx.iteration++
+	path := status.DiskPath
+	if path == "" {
+		path = status.MountPath
+	}
+	log.Infof("handleDiskMetricStatusModify: %s", path)
 }
 
-// Report directory usage for these paths
-var reportDirPaths = []string{
-	types.PersistDir + "/downloads", // XXX old to be removed
-	types.PersistDir + "/img",       // XXX old to be removed
-	types.PersistDir + "/tmp",
-	types.PersistDir + "/log",
-	types.PersistDir + "/rsyslog",
-	types.PersistDir + "/config",
-	types.PersistDir + "/status",
-	types.PersistDir + "/certs",
-	types.PersistDir + "/checkpoint",
+func handleDiskMetricStatusDelete(ctxArg interface{}, key string, statusArg interface{}) {
+	status := statusArg.(types.DiskMetricStatus)
+	ctx := ctxArg.(*zedagentContext)
+	ctx.iteration++
+	path := status.DiskPath
+	if path == "" {
+		path = status.MountPath
+	}
+	log.Infof("handleDiskMetricStatusModify: %s", path)
 }
 
-// Application-related files live here
-// XXX do we need to exclude the ContentTrees used for eve image update?
-// If so how do we tell them apart
-var appPersistPaths = []string{
-	types.VolumeEncryptedDirName,
-	types.VolumeClearDirName,
-	types.SealedDirName + "/downloader",
-	types.SealedDirName + "/verifier",
+func lookupDiskMetricStatus(ctx *zedagentContext, diskPath string) *types.DiskMetricStatus {
```

Comment:
This function doesn't appear to be used.

Do we need to lookup on either diskPath and mountPath? If we can stick to one we can make that the Key and avoid doing a GetAll below.

Reply:
Changed.